src/gallium/drivers/lima/ir/gp/lower.c

```diff
@@ -424,12 +424,32 @@ static bool gpir_lower_eq_ne(gpir_block *block, gpir_node *node)
    return true;
 }
 
+/*
+ * There is no 'abs' opcode.
+ * abs(a) is lowered to max(a, -a)
+ */
+static bool gpir_lower_abs(gpir_block *block, gpir_node *node)
```

Comment:
Could it be implemented as generic nir lowering function? I think we may need the same for ppir.

Reply:
By taking a look at the pp isa doc it seems that there are other ways to implement abs() in pp, so my first thought is that it wouldn't make sense to try to share this: https://github.com/cwabbott0/mali-isa-docs/blob/master/Utgard-PP.md
Also the data structures are quite different in ppir and gpir, not sure it could be shared easily.